verifier/src/tcb.rs

```diff
@@ -336,4 +418,117 @@ mod tests {
         ];
         assert_eq!(tcb.sgx_tcb_components, components);
     }
+
+    #[parameterized(
+        at_issue_date = {"2023-05-10T13:43:27Z"},
+        one_secend_after_issue_date = {"2023-05-10T13:43:28Z"},
+        just_before_next_update = {"2023-06-09T13:43:26Z"},
+    )]
+    fn tcb_verification(time: &str) {
+        let key = tcb_verifying_key();
+        let tcb_json = include_str!("../data/tests/fmspc_00906ED50000_2023_05_10.json");
+        let raw_tcb = TcbInfoRaw::try_from(tcb_json).expect("Failed to parse raw TCB");
+
+        let time = time.parse::<DateTime>().expect("Failed to parse time");
+
+        assert_eq!(raw_tcb.verify(&key, time).is_ok(), true);
+    }
+
+    #[test]
+    fn fails_before_issue_date() {
+        let key = tcb_verifying_key();
+        let tcb_json = include_str!("../data/tests/fmspc_00906ED50000_2023_05_10.json");
+        let raw_tcb = TcbInfoRaw::try_from(tcb_json).expect("Failed to parse raw TCB");
+
+        let time = "2023-05-10T13:43:26Z"
+            .parse::<DateTime>()
+            .expect("Failed to parse time");
+
+        assert_eq!(raw_tcb.verify(&key, time).is_err(), true);
```

Comment:
This goes back to the `serde_json::Error` ideally I think this should be ensuring that this is returning the not yet valid error, but without `PartialEq` it's a compiler failure.

Reply:
one thing we could do is use `assert_matches!` maybe instead?

Reply:
not a blocker

Reply:
:facepalm: Not sure what I was thinking I knew about assert matches, anyway used `assert!(matches!(` because `assert_matches!` is still nightly and wasn't doing any extra logic on the value.